moa/src/main/java/moa/streams/generators/MixedGenerator.java

```diff
@@ -0,0 +1,191 @@
+/*
+ *    MixedGenerator.java
+ *    Copyright (C) 2016 Instituto Federal de Pernambuco
+ *    @author Paulo Gonçalves (paulogoncalves@recife.ifpe.edu.br)
+ *
+ *    This program is free software; you can redistribute it and/or modify
+ *    it under the terms of the GNU General Public License as published by
+ *    the Free Software Foundation; either version 3 of the License, or
+ *    (at your option) any later version.
+ *
+ *    This program is distributed in the hope that it will be useful,
+ *    but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *    GNU General Public License for more details.
+ *
+ *    You should have received a copy of the GNU General Public License
+ *    along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *    
+ */
+package moa.streams.generators;
+
+import com.github.javacliparser.FlagOption;
+import com.github.javacliparser.IntOption;
+import com.yahoo.labs.samoa.instances.Attribute;
+import com.yahoo.labs.samoa.instances.DenseInstance;
+import com.yahoo.labs.samoa.instances.Instance;
+import com.yahoo.labs.samoa.instances.Instances;
+import com.yahoo.labs.samoa.instances.InstancesHeader;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Random;
+import moa.core.InstanceExample;
+
+import moa.core.ObjectRepository;
+import moa.options.AbstractOptionHandler;
+import moa.streams.InstanceStream;
+import moa.tasks.TaskMonitor;
+
+/**
+ * Creates a stream consisting of two boolean and two numeric attributes and
+ * represents a drift by reversing the drift function. Based on proposal by
+ * "Gama, Joao, et al. "Learning with drift detection." Advances in artificial 
+ * intelligence–SBIA 2004. Springer Berlin Heidelberg, 2004. 286-295."
+ *
```

Comment:
We may add the original explanation to make the code easier to understand:

MIXED. Abrupt concept drift, boolean noise-free examples. Four relevant
attributes, two boolean attributes v,w and two numeric attributes
from [0; 1]. The examples are classified positive if two of three conditions
are satisfied: v,w, y < 0,5 + 0,3  sin(3 PI x). After each context change the
classification is reversed.